File: packages/jsii-pacmak/test/expected.dotnet.jsii-calc/AWS.Jsii.Tests.Calculator/UsesInterfaceWithProperties.cs
using AWS.Jsii.Runtime.Deputy;

namespace AWS.Jsii.Tests.Calculator
{
    [JsiiClass(typeof(UsesInterfaceWithProperties), "jsii-calc.UsesInterfaceWithProperties", "[{\"name\":\"obj\",\"type\":{\"fqn\":\"jsii-calc.IInterfaceWithProperties\"}}]")]
    public class UsesInterfaceWithProperties : DeputyBase
    {
        public UsesInterfaceWithProperties(IIInterfaceWithProperties obj): base(new DeputyProps(new object[]{obj}))
        {
        }

        protected UsesInterfaceWithProperties(ByRefValue reference): base(reference)
        {
        }

        protected UsesInterfaceWithProperties(DeputyProps props): base(props)
        {
        }

        [JsiiProperty("obj", "{\"fqn\":\"jsii-calc.IInterfaceWithProperties\"}")]
        public virtual IIInterfaceWithProperties Obj
        {
            get => GetProperty<IIInterfaceWithProperties>();
        }

        [JsiiMethod("justRead", "{\"primitive\":\"string\"}", "[]")]
        public virtual string JustRead()
        {
            return InvokeMethod<string>(new object[]{});
        }

        [JsiiMethod("writeAndRead", "{\"primitive\":\"string\"}", "[{\"name\":\"value\",\"type\":{\"primitive\":\"string\"}}]")]
        public virtual string WriteAndRead(string value)
        {
            return InvokeMethod<string>(new object[]{value});
        }

        [JsiiMethod("readStringAndNumber", "{\"primitive\":\"string\"}", "[{\"name\":\"ext\",\"type\":{\"fqn\":\"jsii-calc.IInterfaceWithPropertiesExtension\"}}]")]
        public virtual string ReadStringAndNumber(IIInterfaceWithPropertiesExtension ext)
        {
            return InvokeMethod<string>(new object[]{ext});
        }
    }
}

File: packages/jsii-pacmak/test/expected.jsii-calc/java/src/main/java/org/jsii/tests/calculator/UsesInterfaceWithProperties.java
package org.jsii.tests.calculator;
@org.jsii.Jsii(module = org.jsii.tests.calculator.$Module.class, fqn = "jsii-calc.UsesInterfaceWithProperties")
public class UsesInterfaceWithProperties extends org.jsii.JsiiObject {
    protected UsesInterfaceWithProperties(final org.jsii.JsiiObject.InitializationMode mode) {
        super(mode);
    }
    public UsesInterfaceWithProperties(final org.jsii.tests.calculator.IInterfaceWithProperties obj) {
        super(org.jsii.JsiiObject.InitializationMode.Jsii);
        org.jsii.JsiiEngine.getInstance().createNewObject(this, java.util.stream.Stream.of(java.util.Objects.requireNonNull(obj, "obj is required")).toArray());
    }
    public java.lang.String justRead() {
        return this.jsiiCall("justRead", java.lang.String.class);
    }
    public java.lang.String writeAndRead(final java.lang.String value) {
        return this.jsiiCall("writeAndRead", java.lang.String.class, java.util.stream.Stream.of(java.util.Objects.requireNonNull(value, "value is required")).toArray());
    }
    public java.lang.String readStringAndNumber(final org.jsii.tests.calculator.IInterfaceWithPropertiesExtension ext) {
        return this.jsiiCall("readStringAndNumber", java.lang.String.class, java.util.stream.Stream.of(java.util.Objects.requireNonNull(ext, "ext is required")).toArray());
    }
    public org.jsii.tests.calculator.IInterfaceWithProperties getObj() {
        return this.jsiiGet("obj", org.jsii.tests.calculator.IInterfaceWithProperties.class);
    }
}
